src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntryHelper.js

```diff
@@ -83,6 +83,21 @@
      * @description: invoke the Data Importer on all records associated with this Batch
      */
     processBatch: function(component) {
+        var totals = component.get('v.totals');
+        var record = component.get('v.record');
+        if (record.RequireTotalMatch__c) {
+            if (totals.countGifts == 0 && totals.totalGiftAmount == 0) {
+                this.showToast(component, $A.get('$Label.c.PageMessagesError'), $A.get('$Label.c.bgeGridWarningRequiredTotalsExpected'), 'warning');
+                return;
+            }
+            
+            if ((record.Expected_Total_Batch_Amount__c != 0 && totals.totalGiftAmount != record.Expected_Total_Batch_Amount__c) ||
```

Comment:
Also will need to check against the field label/name here to make sure this will work for namespacing too.